x/delegation/types/msg.go

```diff
@@ -24,6 +24,25 @@ func (m *MsgDelegation) ValidateBasic() error {
 	return nil
 }
 
+// new message to delegate asset to operator
+func NewMsgDelegation(assetID, fromAddress, approveSignature, approveSalt string, amountPerOperator []KeyValue) *MsgDelegation {
+	baseInfo := &DelegationIncOrDecInfo{
+		FromAddress:        fromAddress,
+		PerOperatorAmounts: make([]KeyValue, 0, 1),
+	}
+	for _, kv := range amountPerOperator {
+		baseInfo.PerOperatorAmounts = append(baseInfo.PerOperatorAmounts, KeyValue{Key: kv.Key, Value: kv.Value})
+	}
+	return &MsgDelegation{
+		AssetID:  assetID,
+		BaseInfo: baseInfo,
+		ApprovedInfo: &DelegationApproveInfo{
```

Comment:
This doesn't seem necessary because it is not used.

Reply:
removed